services/motion/builtin/builtin_test.go

```diff
@@ -494,6 +506,92 @@ func TestObstacleReplanning(t *testing.T) {
 	}
 }
 
+func TestObstacleReplanningSlam(t *testing.T) {
+	cameraToBase := spatialmath.NewPose(r3.Vector{0, 0, 0}, &spatialmath.OrientationVectorDegrees{OY: 1, Theta: -90})
+	cameraToBaseInv := spatialmath.PoseInverse(cameraToBase)
+
+	ctx := context.Background()
+	origin := spatialmath.NewPose(
+		r3.Vector{X: -0.99503e3, Y: 0, Z: 0},
+		&spatialmath.OrientationVectorDegrees{OZ: 1, Theta: -90},
+	)
+
+	boxWrld, err := spatialmath.NewBox(
+		spatialmath.NewPoseFromPoint(r3.Vector{X: 0, Y: 0, Z: 0}),
+		r3.Vector{X: 50, Y: 50, Z: 50}, "box-obstacle",
+	)
+	test.That(t, err, test.ShouldBeNil)
+
+	kb, ms := createMoveOnMapEnvironment(
+		ctx, t,
+		"pointcloud/cardboardOcto.pcd",
+		50, origin,
+	)
+	defer ms.Close(ctx)
+
+	visSrvc, ok := ms.(*builtIn).visionServices[vision.Named("test-vision")].(*inject.VisionService)
+	test.That(t, ok, test.ShouldBeTrue)
+	i := 0
+	visSrvc.GetObjectPointCloudsFunc = func(ctx context.Context, cameraName string, extra map[string]interface{}) ([]*viz.Object, error) {
+		if i == 0 {
+			i++
+			return []*viz.Object{}, nil
+		}
+		currentPif, err := kb.CurrentPosition(ctx)
+		test.That(t, err, test.ShouldBeNil)
+
+		relativeBox := boxWrld.Transform(spatialmath.PoseInverse(currentPif.Pose())).Transform(cameraToBaseInv)
+		detection, err := viz.NewObjectWithLabel(pointcloud.New(), "test-case-1-detection", relativeBox.ToProtobuf())
+		test.That(t, err, test.ShouldBeNil)
+
+		return []*viz.Object{detection}, nil
+	}
+
+	obstacleDetectorSlice := []motion.ObstacleDetectorName{
+		{VisionServiceName: vision.Named("test-vision"), CameraName: camera.Named("test-camera")},
+	}
+	req := motion.MoveOnMapReq{
+		ComponentName: base.Named("test-base"),
+		Destination:   spatialmath.NewPoseFromPoint(r3.Vector{X: 0.8e3, Y: 0, Z: 0}),
```

Comment:
"0.8e3" is both more characters and IMO less legible than "800"

Reply:
will change